website/views/donations.py

```diff
@@ -119,8 +121,24 @@ def webapp_view_donation():
         log(f"Exception when getting a campaign for a donations: {e}", severity="ERROR")
         return render_template("errors/403.html"), 403
 
+    logs_url = None
+    try:
+        trace_id = request.args.get("trace_id")
```

Comment:
Not sure how this gets the trace_id, it's not injected as a query string parameter? You can see how trace is extracted in https://github.com/GoogleCloudPlatform/emblem/blob/main/website/middleware/logging.py
and https://github.com/GoogleCloudPlatform/emblem/blob/32cb6756db291407245918f14e1f900473fd92db/website/middleware/auth.py

Reply:
See line 106 - the `POST /donate` trace ID is captured there, and forwarded to the `GET /viewDonation` request via this query parameter.

(As explained [here](https://github.com/GoogleCloudPlatform/emblem/issues/625#issuecomment-1218669028), the trace ID from `POST /donate` does **not** get propagated to `GET /viewDonation`. Thus, they have _totally separate_ trace IDs.)

Reply:
We can talk more broadly about the trace propagation in #625.

For this specific mode on trace propagation, I don't like the idea that the trace_id is a visible part of the website interface. This may make more sense as session data. @muncus do you have any thoughts on "manual" trace propagation across a web redirect?

Reply:
@grayside cookies seem appropriate here. @engelke are we OK using those instead of session buckets?

(I'd rather not write to GCS with each request. 🙂 )

Reply:
Is there a specific reason the two requests *should* share a trace id? My understanding is that traces are intended to model single interactions, and trace IDs are not intended to be reused between requests (hence why you're having a hard time doing so 😄 ).

If displaying the donation really must be in the same trace, i'd suggest rendering the template here directly, instead of redirecting. Otherwise i'd let the two actions remain separate.

Reply:
@muncus this is useful context, thanks. We could include **both** trace IDs in the log filter, perhaps?